### ISSUE TITLE
Allow Downloader Selection through Argument

### ISSUE THREAD

**Issue by Trenly** [microsoft/winget-cli#2258]
### Description of the new feature / enhancement

As a power user, I would like to be able to select which delivery method is used for downloading installer files on a per-command basis, without having to modify my settings. This will allow for troubleshooting download errors and network stability more easily.

### Proposed technical implementation details

Add a flag for `--delivery-optimization / --do` and a flag for `--wininet' to the install, import, and upgrade commands to control which delivery method is used for downloading installers

**Comment by vedantmgoyal2009** [comment]
I think one flag would be better: `--delivery-method [do | wininet]`

**Comment by Trenly** [comment]
> I think one flag would be better: `--delivery-method [do | wininet]`

I disagree; While a standard argument would work, the proposed multiple-flags would follow the same semantics as the visual style `--rainbow / --retro / --no-vt` and would be easier for users to input, since it wouldn't require the long text of `delivery-method`
